src/useBluetoothNotifications.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+  BluetoothNotificationsHookOptions,
+  BluetoothConnections,
+  BluetoothNotificationsStatus,
+  BluetoothEvent,
+  USER_CANCEL_ERROR_CODE,
+} from "./types/Bluetooth";
+import { useCallback, useState, useEffect, useMemo } from "react";
+import { getFloatValue } from "./utils/getFloatValue";
+
+export const defaultHookOptions: BluetoothNotificationsHookOptions = {
+  serviceUuid: "",
+  characteristicUuid: "",
+  deviceOptions: { acceptAllDevices: true, optionalServices: [] },
+  parser: getFloatValue,
+  onNotification: (parsed: number | string, event: Event) => {
+    return;
+  },
+  onServerDisconnect: (event: Event) => {
+    return;
+  },
+};
+
+export function useBluetoothNotifications(
+  _options: BluetoothNotificationsHookOptions = defaultHookOptions,
+): BluetoothConnections {
+  const [device, setDevice] = useState<BluetoothDevice>();
+  const [server, setServer] = useState<BluetoothRemoteGATTServer>();
+  const [service, setService] = useState<BluetoothRemoteGATTService>();
+  const [characteristic, setCharacteristic] = useState<BluetoothRemoteGATTCharacteristic>();
+  const [status, setStatus] = useState<BluetoothNotificationsStatus>(BluetoothNotificationsStatus.READY);
+
+  const {
+    serviceUuid,
+    characteristicUuid,
+    deviceOptions,
+    onNotification,
+    onServerDisconnect,
+    onError,
+    parser,
+  } = useMemo(() => {
+    return { ...defaultHookOptions, ..._options };
+  }, [_options]);
+
+  const bluetooth: Bluetooth = useMemo(() => {
+    if (!navigator.bluetooth) {
+      const error = new Error("This device is not capable of using bluetooth");
+      if (onError) {
+        onError(error);
+      } else {
+        throw error;
+      }
+    }
```

Comment:
we have handleError now.  use that here?

Reply:
Feelings on using `invariant` for something like this? I believe doing so allows for extra logging features

Reply:
sure invariant seems fine.  But I do find myself wondering now if we should just require the onError prop? are we setting up consumers of this library for runtime exceptions by making this optional?

Reply:
If they are using something like a fallback `ErrorBoundary` to catch all `thrown` errors, would having a way to allow the Boundary to catch it instead be desired?